File: src/main/java/com/lmp/portal/dto/DashboardStatsResponse.java
package com.lmp.portal.dto;

import java.math.BigDecimal;
import java.util.List;

/**
 * DTO de réponse pour les statistiques du tableau de bord utilisateur.
 */
public record DashboardStatsResponse(
        long totalOrders,
        long completedOrders,
        long inProgressOrders,
        long totalReviews,
        long upcomingAppointments,
        BigDecimal totalSpent,
        List<RecentOrderDto> recentOrders,
        List<RecentReviewDto> recentReviews,
        List<UpcomingAppointmentDto> upcomingAppointmentsList
) {

    public record RecentOrderDto(
            String id,
            String serviceName,
            String status,
            BigDecimal totalAmount,
            String currency,
            String createdAt
    ) {}

    public record RecentReviewDto(
            String id,
            int rating,
            String comment,
            boolean approved,
            String createdAt
    ) {}

    public record UpcomingAppointmentDto(
            String id,
            String subject,
            String status,
            String appointmentDate,
            int durationMinutes
    ) {}
}

File: backend-dotnet/src/Lmp.Application/Portal/DashboardStatsResponse.cs
namespace Lmp.Application.Portal;

/// <summary>User dashboard statistics. Faithful port of <c>DashboardStatsResponse</c>.</summary>
public sealed record DashboardStatsResponse(
    long TotalOrders,
    long CompletedOrders,
    long InProgressOrders,
    long TotalReviews,
    long UpcomingAppointments,
    decimal TotalSpent,
    IReadOnlyList<DashboardStatsResponse.RecentOrderDto> RecentOrders,
    IReadOnlyList<DashboardStatsResponse.RecentReviewDto> RecentReviews,
    IReadOnlyList<DashboardStatsResponse.UpcomingAppointmentDto> UpcomingAppointmentsList)
{
    public sealed record RecentOrderDto(
        string Id,
        string? ServiceName,
        string? Status,
        decimal TotalAmount,
        string? Currency,
        string? CreatedAt);

    public sealed record RecentReviewDto(
        string Id,
        int Rating,
        string? Comment,
        bool Approved,
        string? CreatedAt);

    public sealed record UpcomingAppointmentDto(
        string Id,
        string Subject,
        string? Status,
        string? AppointmentDate,
        int DurationMinutes);
}
